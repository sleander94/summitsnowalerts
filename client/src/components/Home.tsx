import { Link } from 'react-router-dom';
import { AuthProps } from '../types.d';

const Home = ({ user }: AuthProps) => {
  return (
    <section id="home">
      <div className="background-image"></div>
      <h1>Summit Snow Alerts</h1>
      <div className="content">
        <h2>
          Never miss a storm. Get text notifications for your favorite
          mountains.
        </h2>
        {user && <p>Welcome back, {user.name.split(' ')[0]}.</p>}
        {!user && (
          <div className="link-container">
            <Link to="/signup">Sign Up</Link>
            <Link to="/login">Log In</Link>
          </div>
        )}
        {user && (
          <div className="link-container">
            <Link to="/conditions">My Mountains</Link>
            <Link to="/account">Edit Account</Link>
          </div>
        )}
      </div>
    </section>
  );
};

export default Home;
