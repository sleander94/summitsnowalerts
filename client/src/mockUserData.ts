export const mockUserData = {
  _id: 'aoifh73$%D3g3',
  name: 'Test User',
  email: 'test@email.com',
  phone: '(555)555-5555',
  password: 'aorinferu142431e5e23e',
  notifications: {
    text: true,
    email: false,
    times: {
      0: true,
      1: false,
      2: false,
      3: false,
      4: false,
      5: false,
      6: false,
      7: false,
      8: true,
      9: false,
      10: false,
      11: false,
      12: false,
      13: false,
      14: false,
      15: false,
      16: false,
      17: false,
      18: true,
      19: false,
      20: false,
      21: false,
      22: false,
      23: false,
    },
    days: {
      sunday: true,
      monday: false,
      tuesday: false,
      wednesday: false,
      thursday: true,
      friday: true,
      saturday: true,
    },
  },
  mountains: { Breckenridge: 80424, Keystone: 80435 },
};
