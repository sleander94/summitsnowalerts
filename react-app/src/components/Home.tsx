import { Link } from 'react-router-dom';
import { AuthProps } from '../types.d';

const Home = ({ user }: AuthProps) => {
  return (
    <section id="home">
      <div className="background-image"></div>
      <h1>Summit Snow Alerts</h1>
      <div className="content">
        <h2>Never miss powder at your favorite mountains</h2>
        {user && <p>Welcome back, {user.name}.</p>}
        {!user && (
          <div className="link-container">
            <Link to="/signup">Sign Up</Link>
            <Link to="/login">Log In</Link>
          </div>
        )}
        {user && (
          <div className="link-container">
            <Link to="/conditions">My Mountains</Link>
            <Link to="/account">Edit Account</Link>
          </div>
        )}
      </div>
    </section>
  );
};

export default Home;
