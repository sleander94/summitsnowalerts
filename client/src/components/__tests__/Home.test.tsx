import Home from '../Home';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mockUserData } from '../../mockUserData';
import '@testing-library/jest-dom/extend-expect';

it('Renders with signup & login buttons when user is undefined', () => {
  const component = render(
    <Router>
      <Home user={undefined} />
    </Router>
  );

  expect(screen.getByText('Sign Up')).toHaveAttribute('href', '/signup');
  expect(screen.getByText('Log In')).toHaveAttribute('href', '/login');

  expect(component).toMatchSnapshot();
});

it('Renders with forecast & alerts buttons & greeting when user is logged in', () => {
  const component = render(
    <Router>
      <Home user={mockUserData} />
    </Router>
  );

  expect(screen.getByText('Welcome back, Test.'));
  expect(screen.getByText('Forecast')).toHaveAttribute('href', '/forecast');
  expect(screen.getByText('Alerts')).toHaveAttribute('href', '/alerts');

  expect(component).toMatchSnapshot();
});
