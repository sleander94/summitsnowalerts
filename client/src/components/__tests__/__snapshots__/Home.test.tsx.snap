// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with forecast & alerts buttons & greeting when user is logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="home"
      >
        <div
          class="background-image"
        />
        <div
          class="content"
        >
          <h1>
            Summit Snow Alerts
          </h1>
          <h2>
            Never miss a storm. Get text notifications for your favorite mountains.
          </h2>
          <p>
            Welcome back, 
            Test
            .
          </p>
          <div
            class="link-container"
          >
            <a
              href="/forecast"
            >
              Forecast
            </a>
            <a
              href="/alerts"
            >
              Alerts
            </a>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="home"
    >
      <div
        class="background-image"
      />
      <div
        class="content"
      >
        <h1>
          Summit Snow Alerts
        </h1>
        <h2>
          Never miss a storm. Get text notifications for your favorite mountains.
        </h2>
        <p>
          Welcome back, 
          Test
          .
        </p>
        <div
          class="link-container"
        >
          <a
            href="/forecast"
          >
            Forecast
          </a>
          <a
            href="/alerts"
          >
            Alerts
          </a>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders with signup & login buttons when user is undefined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="home"
      >
        <div
          class="background-image"
        />
        <div
          class="content"
        >
          <h1>
            Summit Snow Alerts
          </h1>
          <h2>
            Never miss a storm. Get text notifications for your favorite mountains.
          </h2>
          <div
            class="link-container"
          >
            <a
              href="/signup"
            >
              Sign Up
            </a>
            <a
              href="/login"
            >
              Log In
            </a>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="home"
    >
      <div
        class="background-image"
      />
      <div
        class="content"
      >
        <h1>
          Summit Snow Alerts
        </h1>
        <h2>
          Never miss a storm. Get text notifications for your favorite mountains.
        </h2>
        <div
          class="link-container"
        >
          <a
            href="/signup"
          >
            Sign Up
          </a>
          <a
            href="/login"
          >
            Log In
          </a>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
